src/Internal/DependencyManagement/GoogleServiceProvider.php

```diff
@@ -192,6 +193,14 @@ protected function error_handler(): callable {
 				return $handler( $request, $options )->then(
 					function ( ResponseInterface $response ) use ( $request ) {
 						$code = $response->getStatusCode();
+
+						$path = $request->getUri()->getPath();
+
+						// Partial Failures come back with a status code of 200, so it's necessary to call GoogleAdsFailures:init every time.
+						if ( strpos( $path, 'google-ads' ) !== false ) {
+							GoogleAdsFailures::init();
```

Comment:
I was wondering what happens if we ever send two requests (one after the other). However it seems that the init function only initializes a new instance if it can't find a previously initialized one.

Reply:
Yes that's correct.